e2e/opts.go

```diff
@@ -0,0 +1,42 @@
+package e2e_test
+
+import (
+	"fmt"
+
+	"github.com/Azure/agentbaker/pkg/agent/datamodel"
+	"github.com/Azure/agentbakere2e/scenario"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/containerservice/armcontainerservice"
+)
+
+var (
+	errNilClusterModelProperties = fmt.Errorf("cluster model or cluster model properties were nil")
+)
+
+type scenarioRunOpts struct {
+	cloud         *azureClient
+	kube          *kubeclient
+	suiteConfig   *suiteConfig
+	scenario      *scenario.Scenario
+	chosenCluster *armcontainerservice.ManagedCluster
+	nbc           *datamodel.NodeBootstrappingConfiguration
+	subnetID      string
+	loggingDir    string
+}
+
+// Returns true if the scenario's chosen cluster is configured with Azure CNI
+func (opts *scenarioRunOpts) isChosenClusterAzureCNI() (bool, error) {
+	cluster := opts.chosenCluster
+	if cluster != nil && cluster.Properties != nil && cluster.Properties.NetworkProfile != nil {
```

Comment:
move the whole condition to a helper function with descriptive name? I think that would help with readability.